seed/static/seed/js/controllers/insights_property_controller.js

```diff
@@ -31,49 +31,93 @@ angular.module('BE.seed.controller.insights_property', [])
+          }
+        }
+
+        // else use first compliance_metric
+        if (compliance_metrics.length > 0) {
+          $scope.configs.compliance_metric = compliance_metrics[0];
+          $scope.selected_metric = $scope.configs.compliance_metric.id;
+        }
       }
+      set_init_compliance_metric(saved_configs);
 
       // chart data
       $scope.data = null;
       $scope.chart_datasets = {};
 
       // default settings / dropdowns
-      $scope.chart_cycle = null;
       $scope.chart_cycle_name = null;
       $scope.cycles = [];
-      $scope.chart_metric = null;
-      $scope.chart_xaxis = null;
       $scope.x_axis_options = [];
       $scope.y_axis_options = [];
       $scope.x_categorical = false;
 
+      $scope.$watch('configs', function (new_configs) {
+        const local_storage_configs = {
+          compliance_metric_id: new_configs.compliance_metric.id,
+          chart_cycle: new_configs.chart_cycle,
+          chart_metric: new_configs.chart_metric,
+          chart_xaxis: new_configs.chart_xaxis,
+          dataset_visibility: new_configs.dataset_visibility,
+          annotation_visibility: new_configs.annotation_visibility,
+        }
+        localStorage.setItem('insights.property.configs',  JSON.stringify(local_storage_configs));
+      }, true);
+
       // load data
       let _load_data = function () {
-        if (_.isEmpty($scope.compliance_metric)) {
+        if (_.isEmpty($scope.configs.compliance_metric)) {
           spinner_utility.hide();
           return;
         }
         spinner_utility.show();
-        compliance_metric_service.evaluate_compliance_metric($scope.compliance_metric.id).then((data) => {
+        compliance_metric_service.evaluate_compliance_metric($scope.configs.compliance_metric.id).then((data) => {
           $scope.data = data;
         }).then(() => {
           if ($scope.data) {
             // set options
             // cycles
             $scope.cycles = $scope.data.metric.cycles;
             if (_.size($scope.cycles) > 0){
+              // used saved cycle
+              if(saved_configs && saved_configs.hasOwnProperty("chart_cycle")){
+                const saved_cycle = $scope.cycles.find(c => c.id == saved_configs.chart_cycle);
+                if (saved_cycle !== undefined){
+                  $scope.configs.chart_cycle = saved_cycle.id;
+                  $scope.chart_cycle_name = saved_cycle.name;
+                }
+                delete saved_configs["chart_cycle"] // don't trigger this if again, only the first time.
+              }
+
               // don't clear out a valid existing selection
-              if($scope.chart_cycle == null || _.find($scope.cycles, function(o) { return o.id == $scope.chart_cycle}) == undefined) {
-                $scope.chart_cycle = _.first($scope.cycles).id
+              if($scope.configs.chart_cycle == null || _.find($scope.cycles, function(o) { return o.id == $scope.configs.chart_cycle}) == undefined) {
```

Comment:
All comparisons, especially to `null` or `undefined`, need to use `===` to prevent type coercion.

Simplify: `!$scope.configs.chart_cycle || !$scope.cycles.find(({id}) => id === $scope.configs.chart_cycle)`